dockerfiles/cpubuilder_ubuntu_jammy_x86_64.Dockerfile

```diff
@@ -21,22 +22,33 @@ RUN echo "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-${LLVM_VERSION} ma
     curl https://apt.llvm.org/llvm-snapshot.gpg.key | gpg --dearmor > /etc/apt/trusted.gpg.d/llvm-snapshot.gpg && \
     apt update && \
     apt install -y clang-${LLVM_VERSION} lld-${LLVM_VERSION}
+# Virtual file system adapter for Azure Blob storage, used for remote cache.
+RUN curl -sSL -O https://packages.microsoft.com/config/ubuntu/22.04/packages-microsoft-prod.deb
+RUN dpkg -i packages-microsoft-prod.deb
+RUN rm packages-microsoft-prod.deb
+RUN apt-get update && \
+    apt-get install -y fuse3 blobfuse2
 # Cleanup.
 RUN apt clean && rm -rf /var/lib/apt/lists/*
 
+######## CCache ########
+WORKDIR /install-ccache
+COPY build_tools/install_ccache.sh ./
+RUN ./install_ccache.sh "4.10.2" && rm -rf /install-ccache
+
 ######## CMake ########
 WORKDIR /install-cmake
 ENV CMAKE_VERSION="3.23.2"
 COPY build_tools/install_cmake.sh ./
 RUN ./install_cmake.sh "${CMAKE_VERSION}" && rm -rf /install-cmake
 
 ######## Build toolchain configuration ########
-# Setup symlinks and alternatives then default to using clang-14.
-# This can be overriden to gcc or another clang version as needed.
+# Setup symlinks and alternatives.
 RUN ln -s /usr/bin/lld-14 /usr/bin/lld && \
     ln -s /usr/bin/ld.lld-14 /usr/bin/ld.lld && \
     ln -s /usr/bin/clang-14 /usr/bin/clang && \
     ln -s /usr/bin/clang++-14 /usr/bin/clang++
+# Default to using clang. This can be overriden to gcc as desired.
 ENV CC=clang
```

Comment:
Some diffs slipped in here while rebasing. Can you clean those up? (Generally, doing a self-review on your PR can help catch these)